otter/convergence/service.py

```diff
@@ -77,7 +82,8 @@ def update_group_state(group, old_state):
 
 
 def execute_convergence(tenant_id, group_id, log,
-                        get_all_convergence_data=get_all_convergence_data):
+                        get_all_convergence_data=get_all_convergence_data,
+                        plan=plan):
```

Comment:
Good idea. It grew from a very simple function and I kept adding callbacks, now it definitely seems @do would be better.